AzureRM Terraform Backend (#2)
redhat-cop/cloud.terraform_ops: roles/azurerm_backend/README.md

# azurerm_backend

> A role to ensure that the necessary AWS infrastructure is present/absent for an S3 remote backend for Terraform.

Reviewer: ```suggestion
A role to ensure that the necessary Azure infrastructure is present/absent for an storage account remote backend for Terraform.
```

> When creating, the S3 bucket will be created with the required permissions for Terraform. See U(https://developer.hashicorp.com/terraform/language/settings/backends/s3#s3-bucket-permissions).

Reviewer: ```suggestion
When creating, the storage account will be created with the required permissions for Terraform. See U(https://developer.hashicorp.com/terraform/language/settings/backends/s3#s3-bucket-permissions).
```

> The role also allow for optionally creating a DynamoDB table with the required permissions for state locking and with a partition key named LockID with type of String.

Reviewer: ```suggestion
The role also allow for optionally creating a with the required permissions for state locking and with a partition key named LockID with type of String.
```
some places have leftover AWS terminologies and references in Azure backend, DynamoDB should be storage account I think

# azurerm_backend

A role to create/delete the necessary Azure infrastructure for an Azurerm remote backend for Terraform.
The role ensures that the specified resource group, storage account and container are present/absent. When a service principal is specified, the role will assign the [Storage Blob data contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor) to the service principal.

## Requirements

Azure Account with permission to create Resource group, Storage account and container, and eventually assign role to service principal.

## Role Variables

Name | Description | Type | Default | Required
--- | --- | --- | --- | ---

> azurerm_backend_operation|Whether to create or delete the infrastructure for the azurerm Terraform backend.. Choices: 'create', 'delete'|string|'create'|N/A

Reviewer 2: ```suggestion
azurerm_backend_operation|Whether to create or delete the infrastructure for the azurerm Terraform backend. Choices: 'create', 'delete'|string|'create'|N/A
```

azurerm_backend_container_name|The Name of the Storage Container to create within the Storage Account.|string|N/A|When __azurerm_backend_operation=create__.
azurerm_backend_service_principal_id|The service principal Id used by Terraform to push Terraform state to the container. The role will assign the [Storage Blob data contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor) role for the specified container to this service principal.|string|N/A|No

## Example Playbook

    - hosts: localhost
      roles:
        - role: cloud.terraform_ops.azurerm_backend
          azurerm_backend_operation: create
          azurerm_backend_resource_group_name: "StorageAccount-ResourceGroup"
          azurerm_backend_location: eastus
          azurerm_backend_storage_account_name: ansible123
          azurerm_backend_storage_account_type: Premium_LRS
          azurerm_backend_container_name: tfstate
          azurerm_backend_service_principal_id: abcdef12-123a-456b-789c-12345abcde6e

## License

GNU General Public License v3.0 or later

> See [LICENCE](https://github.com/ansible-collections/cloud.terraform_ops/blob/main/LICENSE) to see the full text.

Reviewer 2: ```suggestion
See [LICENCE](https://github.com/redhat-cop/cloud.terraform_ops/blob/main/LICENSE) to see the full text.
```
